spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/DefaultBinding.java

```diff
@@ -64,9 +71,21 @@ public String getName() {
 	public String getGroup() {
 		return this.group;
 	}
+	
+	@Override
+	public final synchronized void bind() {
+		if (this.lifecycle != null && !this.lifecycle.isRunning()) {
+			if (StringUtils.hasText(this.group)) {
+				this.lifecycle.start();
+			}
+			else {
+				logger.warn("Can not re-bind an anonymous binding");
```

Comment:
Won't this happen at start up time? Or do you mean that `group` is not empty there yet, but it is cleaned afterwards ?

Reply:
see the comment on the interface. It's all a bit awkward but at the moment I have to deal with what I have and in this case it's only to call IF and only IF you called `unbind` before, otherwise its idempotent